glib/src/log.rs

```diff
@@ -114,27 +114,39 @@ fn to_log_flags(fatal: bool, recursion: bool) -> u32 {
 }
 
 #[cfg(any(feature = "v2_46", feature = "dox"))]
-pub fn log_set_handler<P: Fn(&str, LogLevel, &str) + Send + Sync + 'static>(
+pub fn log_set_handler<P: Fn(Option<&str>, LogLevel, &str) + Send + Sync + 'static>(
     log_domain: Option<&str>,
     log_levels: LogLevels,
     fatal: bool,
     recursion: bool,
     log_func: P,
 ) -> LogHandlerId {
     let log_func_data: Box_<P> = Box_::new(log_func);
-    unsafe extern "C" fn log_func_func<P: Fn(&str, LogLevel, &str) + Send + Sync + 'static>(
+    unsafe extern "C" fn log_func_func<
+        P: Fn(Option<&str>, LogLevel, &str) + Send + Sync + 'static,
+    >(
         log_domain: *const libc::c_char,
         log_level: glib_sys::GLogLevelFlags,
         message: *const libc::c_char,
         user_data: glib_sys::gpointer,
     ) {
-        let log_domain: Borrowed<GString> = from_glib_borrow(log_domain);
         let message: Borrowed<GString> = from_glib_borrow(message);
         let callback: &P = &*(user_data as *mut _);
-        (*callback)(log_domain.as_str(), from_glib(log_level), message.as_str());
+        if log_domain.is_null() {
```

Comment:
You can do `let log_domain: Borrowed<Option<GString>> = from_glib_borrow(log_domain)` and then use it as `log_domain.as_deref()` or similar

Reply:
No need for the separate `NULL` check

Reply:
I've fixed that in the next commit.